doc/sphinx/proof-engine/tactics.rst

```diff
@@ -105,6 +108,354 @@ bindings_list`` where ``bindings_list`` may be of two different forms:
+
+.. productionlist::        coq
+   intropatterns           : `intropattern` ... `intropattern`
+                           : `empty`
+   empty                   :
+   intropattern            : *
+                           : **
+                           : `simple_intropattern`
+   simple_intropattern     : `simple_intropattern_closed` [ % `term` ... % `term` ]
+   simple_intropattern_closed : `naming_intropattern`
+                              : _
+                              : `or_and_intropattern`
+                              : `equality_intropattern`
+   naming_intropattern     : `ident`
+                           : ?
+                           : ?`ident`
+   or_and_intropattern     : [ `intropatterns` | ... | `intropatterns` ]
+                           : ()
+                           : ( `simple_intropattern` , ... , `simple_intropattern` )
+                           : ( `simple_intropattern` & ... & `simple_intropattern` )
+   equality_intropattern   : ->
+                           : <-
+                           : [= `intropatterns` ]
+   or_and_intropattern_loc : or_and_intropattern
+                           : `ident`
+
+Note that the intro pattern syntax varies between tactics.
+Most tactics use :n:`@simple_intropattern` in the grammar.
+:tacn:`destruct`, :tacn:`edestruct`, :tacn:`induction`,
+:tacn:`einduction`, :tacn:`case`, :tacn:`ecase` and the various
+:tacn:`inversion` tactics use :n:`@or_and_intropattern_loc`, while
+:tacn:`intros` and :tacn:`eintros` use :n:`@intropatterns`.
+The :n:`eqn:` construct in various tactics uses :n:`@naming_intropattern`.
+
+**Naming patterns**
+
+Use these elementary patterns to specify a name:
+
+* :n:`@ident` - use the specified name
+* :n:`?` - let Coq choose a name
+* :n:`?@ident` - generate a name that begins with :n:`@ident`
+* :n:`_` - discards the matched part (unless it is required for another
+  hypothesis)
+* if a disjunction pattern omits a name, such as :g:`[|H2]`, Coq will choose a name
+
+**Splitting patterns**
+
+The most common splitting patterns are:
+
+* split a hypothesis in the form :n:`A /\ B` into two
+  hypotheses :g:`H1: A` and :g:`H2: B` using the pattern :g:`(H1 & H2)` or
+  :g:`(H1, H2)` or :g:`[H1 H2]`
+  :ref:`Example <intropattern_conj_ex>`
+* split a hypothesis in the form :g:`A \/ B` into two
+  subgoals using the pattern :g:`[H1|H2]`.  The first subgoal will have the hypothesis
+  :g:`H1: A` and the second subgoal will have the hypothesis :g:`H2: B`.
+  :ref:`Example <intropattern_disj_ex>`
+* split a hypothesis in either of the forms :g:`A /\ B` or :g:`A \/ B` using the pattern :g:`[]`.
+
+**A node has to be of type Prop to be matched/split/named, right?**
+
+**what other common operators are intro patterns useful for?  ->? <->? forall?  I would like to
+mention them**
+
+Patterns can be nested: :n:`[[Ha|Hb] H]` can be used to split :n:`(A \/ B) /\ C`.
+
+Note that there is no equivalent to intro patterns for goals.  For a goal :g:`A /\ B`,
```

Comment:
Strictly speaking, a goal is the hypotheses plus the conclusion, right?  The same as "proof context"?  "proof context" or "context" seems clearer than "goal".  References such as "For a goal :g:`A /\ B`" on this line seem very natural.  Or do we want to be strict and always use the term"conclusion"?

Reply:
Hum. Strictly speaking, we may include the context in what we call "a goal" (and this is what `Show Goal` does). But when we speak more specifically of what appears below the bar, we generally say "goal" as well. Here, I meant the conclusion of the latter, i.e. what goes after the last `->`

Reply:
Huh?  Where is the `->` here?  "conclusion" seems out of place.  I'm talking about applying `split` in this case:

```
1 subgoal
A, B : Prop
______________________________________(1/1)
A /\ B
```
The paragraph is not about intro patterns per se; it's meant to provide clarifying contrast for when you have `A /\ B` below the line.  Perhaps it would be better if the first sentence was "Note that you must use other tactics to split up goals."

Or I could drop the paragraph entirely.

Reply:
The conclusion is the full goal because there is no `->`. Anyway, it seems that the "clarification" I proposed wasn't one in reality. I prefer that we keep your paragraph as is. I think it is really useful.